src/oscar/apps/basket/forms.py

```diff
@@ -293,9 +350,11 @@ def cleaned_options(self):
         options = []
         for option in self.parent_product.options:
             if option.code in self.cleaned_data:
-                options.append({
-                    'option': option,
-                    'value': self.cleaned_data[option.code]})
+                value = self.cleaned_data[option.code]
+                if option.required or value not in EMPTY_VALUES:
+                    options.append(
+                        {"option": option, "value": option.clean_value(value)}
```

Comment:
This is also not a backward compatible change - previously we would pass a string value here and this is no longer the case. A value `"foo"` now gets JSON encoded to `'"foo"'` - which would break the logic of any project that is processing option values assuming they are simple strings. It also somewhat messes up the display of the line in the basket etc. 

I think it would be better to JSON-encode only values that are not simple. 

We also need to make sure option/multi option values are displayed in a sensible way in the basket, by default.

Reply:
I'm not too sure about having special cases for simple values is a good idea, especially because they where not handled correctly to begin with. If the option type is int and I receive a string from the database that is not correct behaviour.

So what could be done here, is make the breaking change more apparent by introducing a migration and changing the field type to something that encodes to json before storage. We could then also have a data migration that migrates all existing data. The code that treats string values as simple string would then still work, only when you where treating integers as simple string you would have a problem, which seems correct to me.

How does that sound to you @solarissmoke

Reply:
Sounds good.

Reply:
(Still think we need to improve the default display of these in the basket - rendering a JSON blob as string doesn't feel like a usable default)

Reply:
So I just implemented the JSONField but I'm not sure if I mind the JSON blob as much as you do 😃 
That will take some template changes and I'm not really up for that right now.

Reply:
You mention above that we'd need a data migration to migrate existing (CharField) data to the JSONField - I don't see that data migration in the PR? 

We also need to include a prominent note about this in the release notes. 

I still feel that the change in how these options are displayed by default is problematic, and that if we are going to improve this functionality then it is worth the effort of sorting out the templates at the same time. Whether or not that should block merging of the functional changes is a matter of opinion and I expect that you will argue it's better to merge and improve later. I just think Oscar needs to to better at shipping usable defaults.

Reply:
The data migration is not needed because the existing data in the charfield is valid json. That is why all the tests still pass as well.

We will have an intern mess around with the templates tomorrow @viggo-devries will make that happen.

Reply:
The display of the options has been fixed.